bavard_ml_utils/persistence/record_store/dynamodb.py

```diff
@@ -91,24 +93,53 @@ def delete_all(self, *conditions: t.Tuple[str, str, t.Any], **where_equals) -> i
         # TODO: This is an incredibly slow way of doing this. Use indexes and a batch delete instead.
         self.assert_can_edit()
         num_deleted = 0
-        for record in self.get_all(*conditions, **where_equals):
+        # for record in self.get_all(*conditions, **where_equals):
```

Comment:
Will we be needing this commented out line of code anymore? If not, I think we can delete it.

Reply:
@epeters3 Thanks very much for your review. You are right, we don't need it. I just kept it for now. I also think we don't need to have both _scan and get_all anymore since _scan now is taking care of filtering conditions. So, I think get_all can be deleted and _scan can be renamed to get_all.

Reply:
Good point @Alifilsoofi, thank you. Let's make sure we do that so we can keep the code clean, which makes it easier to maintain. Thanks again for these changes so far.